stl/inc/xstring

```diff
@@ -2642,6 +2642,44 @@ private:
     }
 
 public:
+#if _HAS_CXX20
+    void _Move_assign_from_buffer(_Elem* const _Right, const size_type _Size, const size_type _Res) {
+        // Move assign from a buffer, used by basic_stringbuf
+        _Tidy_deallocate();
+        _Right[_Size]  = _Elem();
+        auto& _My_data = _Mypair._Myval2;
+        _Construct_in_place(_My_data._Bx._Ptr, pointer_traits<pointer>::pointer_to(*_Right));
+        _My_data._Mysize = _Size;
+        _My_data._Myres  = _Res;
+    }
+
+    struct _Released_buffer {
+        _Elem* _Ptr;
+        size_type _Size;
+        size_type _Res;
+    };
+
+    _NODISCARD _Released_buffer _Release_to_buffer() {
+        // Release to a buffer, or allocate a new one if in small string mode
+        _Released_buffer _Result;
+        auto& _My_data = _Mypair._Myval2;
+        if (_My_data._Large_string_engaged()) {
+            _Result._Ptr      = _Unfancy(_My_data._Bx._Ptr);

```

Comment:
Can this thing be used with fancy pointers? If not, you can skip _Unfancy.

Reply:
OK. But anyway the `_Unfancy` will be called. If not in string, then in stringbuf.

Reply:
I mean, `stringbuf` does not support fancy pointers, so I don't think this feature works at all with allocators that use them, so jumping through hoops for them may not be worthwile.